openeo/rest/datacube.py

```diff
@@ -1900,6 +1900,87 @@ def validate(self) -> List[dict]:
     def tiled_viewing_service(self, type: str, **kwargs) -> Service:
         return self._connection.create_service(self.flat_graph(), type=type, **kwargs)
 
+    def _get_spatial_extent_from_load_collection(self):
+        pg = self.flat_graph()
+        for node in pg:
+            if pg[node]["process_id"] == "load_collection":
+                if "spatial_extent" in pg[node]["arguments"] and all(
+                    cd in pg[node]["arguments"]["spatial_extent"]
+                    for cd in ["east", "west", "south", "north"]
+                ):
+                    return pg[node]["arguments"]["spatial_extent"]
+        return None
+
+    def preview(
+        self,
+        center: Union[Iterable, None] = None,
+        zoom: Union[int, None] = None,
+    ):
+        """
+        Creates a service with the process graph and displays a map widget. Only supports XYZ.
+
+        :param center: (optional) Map center. Default is (0,0).
+        :param zoom: (optional) Zoom level of the map. Default is 1.
+
+        :return: ipyleaflet Map object and the displayed Service
+        """
+        if "XYZ" not in self.connection.list_service_types():
+            raise OpenEoClientException("Backend does not support service type 'XYZ'.")
+
+        try:
+            from IPython.core.getipython import get_ipython
+            from IPython.display import display
+            import ipyleaflet
+            
+            # Check we are inside Jupyer Notebook
+            shell = get_ipython().__class__.__name__
+            if shell != "ZMQInteractiveShell":
+                raise Exception("On-demand preview only supported in Jupyter notebooks!")
+        except ImportError:
+            raise Exception(
+                "Additional modules must be installed for on demand viewer. Run `pip install openeo[jupyter]` or refer to the documentation."
+            )
+        except NameError:
+            raise Exception("On-demand preview only supported in Jupyter notebooks!")
```

Comment:
under #237/#420 I added `in_jupyter_context` helper:
https://github.com/Open-EO/openeo-python-client/blob/48edef8256450772a9dd5d30d59ab343b66edfec/openeo/internal/jupyter.py#L85-L92

this should make it possible to simplify this whole try-except construct